internal/cmd/shell_systemd.go

```diff
@@ -0,0 +1,36 @@
+package cmd
+
+import (
+	"errors"
+)
+
+// systemdShell is not a real shell
+type systemdShell struct{}
+
+// Systemd is not really a shell but is useful to add support
+// to systemd EnvironmentFile(https://0pointer.de/public/systemd-man/systemd.exec.html#EnvironmentFile=)
+var Systemd Shell = systemdShell{}
+
+func (sh systemdShell) Hook() (string, error) {
+	return "", errors.New("this feature is not supported")
+}
+
+func (sh systemdShell) Export(e ShellExport) (out string) {
+	for key, value := range e {
+		if value != nil {
+			out += sh.export(key, *value)
+		}
+	}
+	return out
+}
+
+func (sh systemdShell) Dump(env Env) (out string) {
+	for key, value := range env {
+		out += sh.export(key, value)
+	}
+	return out
+}
+
+func (sh systemdShell) export(key, value string) string {
+	return key + "=" + "\"" + value + "\"\n"
```

Comment:
I didn't find an answer for `"`, but this should be handled:
> A line ending with a backslash will be concatenated with the following one

Reply:
So I did a bit of testing locally, and for this `test_env_file`:
```
e1=example
e2="with whitespaces in it and quotes around"
e3=with whitespace but without quotes around it
e4= without quotes and with leading whitespace
e4_2= without quotes and with leading whitespace
e5= without quotes around it but quotes "" in it
e5_2="with quotes around it and quotes "" in "" it"
e5_3="with quotes around it and single quotes '' in '' it'"
e5_4='with single quotes around it and double quotes "" in it"'
e5_5='with single quotes around it and '' single quotes in it '''
e6= line return \n in string without quotes
e7="line return \n in string with quotes around"
e8=multiple line
example
without quotes around
e9=" multiple line
but this time with
quotes around it "
```
`systemd` parses it and outputs:
```
e1=example
e2=with whitespaces in it and quotes around
e3=with whitespace but without quotes around it
e4=without quotes and with leading whitespace
e4_2=without quotes and with leading whitespace
e5=without quotes around it but quotes "" in it
e5_2=with quotes around it and quotes  in  it
e5_3=with quotes around it and single quotes '' in '' it'
e5_4=with single quotes around it and double quotes "" in it"
e5_5=with single quotes around it and  single quotes in it
e6=line return n in string without quotes
e7=line return \n in string with quotes around
e8=multiple line
e9= multiple line
but this time with
quotes around it
```

The conclusion is:
- without `"` or `'` around the value, `systemd` will
  > strips leading and trailing whitespace from the values of assignments
  
  but also removes special character like `\` or newlines character BUT it doesn't remove `"` or `'` .
- with `"` or `'` around the value, `systemd` removes all occurrences of the type of quotes that encapsulate the value `"` or `'`. Otherwise `systemd` doesn't remove any special characters.

Reply:
Yes I will implement this parsing, just having troubles finding the time right now.

Reply:
First implementation is ready for review.
I found out that
> if the value has special characters and also contains both type of quotes in it, then we should log a message at least or maybe even error as we cannot ensure the integrity of the parsed value by systemd.
Is wrong, if we escape the quotes with `\` it should be preserved when parsed by systemd.